### PR TITLE
Allow older CMake version

### PR DESCRIPTION
**Pull request by masterleinad**
At least for `CMake 3.10` we need absolute paths for `INTERFACE` `target_sources`.
Of course, the usual incompatibility(https://stackoverflow.com/questions/42123509/cmake-finds-boost-but-the-imported-targets-not-available-for-boost-version) between `Boost` and `CMake` still applies, i.e. for `CMake 3.10` `Boost` version >1.65.1 don't work with imported targets correctly.

### CODE REVIEW

**Comment by aprokop** [comment]
ok to test

**Comment by dalg24** [comment]
add to whitelist

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -1,32 +1,32 @@
 target_sources(ArborX INTERFACE
```
Comment (dalg24) [inline src/CMakeLists.txt]: I am removing `target_source()` in #21

Comment (masterleinad) [reply src/CMakeLists.txt]: Even better!

**Comment by masterleinad** [comment]
Superseded by #21.

**Pull request closed without merging**